packages/remix-dev/vite/build.ts

```diff
@@ -8,53 +8,15 @@ import {
   type BuildManifest,
   type ServerBundleBuildConfig,
   type ServerBundlesBuildManifest,
+  resolveViteConfig,
+  extractRemixPluginContext,
   configRouteToBranchRoute,
   getServerBuildDirectory,
 } from "./plugin";
 import type { ConfigRoute, RouteManifest } from "../config/routes";
 import invariant from "../invariant";
 import { preloadViteEsm } from "./import-vite-esm-sync";
 
-async function resolveViteConfig({
-  configFile,
-  mode,
-  root,
-}: {
-  configFile?: string;
-  mode?: string;
-  root: string;
-}) {
-  let vite = await import("vite");
-
-  // Leverage the Vite config as a way to configure the entire multi-step build
-  // process so we don't need to have a separate Remix config
-  let viteConfig = await vite.resolveConfig(
-    { mode, configFile, root },
-    "build", // command
-    "production", // default mode
-    "production" // default NODE_ENV
-  );
-
-  if (typeof viteConfig.build.manifest === "string") {
-    throw new Error("Custom Vite manifest paths are not supported");
-  }
-
-  return viteConfig;
-}
-
-async function extractRemixPluginContext(viteConfig: Vite.ResolvedConfig) {
-  let ctx = viteConfig["__remixPluginContext" as keyof typeof viteConfig] as
-    | RemixPluginContext
-    | undefined;
-
-  if (!ctx) {
-    console.error(colors.red("Remix Vite plugin not found in Vite config"));
-    process.exit(1);
-  }
-
-  return ctx;
-}
-
```

Comment:
This has moved to `vite/plugin.ts` since the logic is now shared and not specific to the `vite:build` command.